### PR TITLE
 Make release tools asynchronous

### PR DESCRIPTION
**Pull request by pomek**
### Suggested merge commit message ([convention](https://github.com/ckeditor/ckeditor5-design/wiki/Git-commit-message-convention))

Internal: Make release tools asynchronous. Closes ckeditor/ckeditor5#14135.

### CODE REVIEW

File: packages/ckeditor5-dev-release-tools/lib/tasks/commitandtag.js [hunk 1]
```diff
@@ -16,13 +16,13 @@ const { globSync } = require( 'glob' );
  * @param {String} options.version The commit will contain this param in its message and the tag will have a `v` prefix.
  * @param {Array.<String>} options.files Array of glob patterns for files to be added to the release commit.
  * @param {String} [options.cwd=process.cwd()] Current working directory from which all paths will be resolved.
+ * @returns {Promise<Void>}
  */
-module.exports = function commitAndTag( { version, files, cwd = process.cwd() } ) {
+module.exports = async function commitAndTag( { version, files, cwd = process.cwd() } ) {
 	const normalizedCwd = toUnix( cwd );
 	const filePathsToAdd = globSync( files, { cwd: normalizedCwd, absolute: true, nodir: true } );
```
Comment (pomek) [inline packages/ckeditor5-dev-release-tools/lib/tasks/commitandtag.js:23]: Expected to be async.

File: packages/ckeditor5-dev-release-tools/lib/tasks/updatedependencies.js
Comment (pomek) [inline packages/ckeditor5-dev-release-tools/lib/tasks/updatedependencies.js:1]: L59 - expected to be async.

**Review by psmyrek — APPROVED** [review]
LGTM 🎉

File: packages/ckeditor5-dev-release-tools/lib/utils/executeinparallel.js [hunk 2]
```diff
@@ -26,65 +25,80 @@ const WORKER_SCRIPT = upath.join( __dirname, 'parallelworker.js' );
...
 	} );
 	const packagesInThreads = getPackagesGroupedByThreads( packages, concurrency );
 
 	const callbackModule = upath.join( cwd, crypto.randomUUID() + '.js' );
-	fs.writeFileSync( callbackModule, `'use strict';\nmodule.exports = ${ taskToExecute };`, 'utf-8' );
+	await fs.writeFile( callbackModule, `'use strict';\nmodule.exports = ${ taskToExecute };`, 'utf-8' );
 
-	const counter = tools.createSpinner( processDescription, { total: packages.length } );
-	counter.start();
+	const onPackageDone = progressFactory( listrTask, packages.length );
```
Comment (martnpaneq) [inline packages/ckeditor5-dev-release-tools/lib/utils/executeinparallel.js:56]: Looks like `progressFactory` could be passed as an argument to `executeInParallel`. This way `executeInParallel` would be independent of the task runner if we chose to use a different one in the future for other use cases.

File: packages/ckeditor5-dev-release-tools/tests/tasks/updatedependencies.js [hunk 3]
```diff
@@ -157,30 +157,32 @@ describe( 'dev-release-tools/tasks', () => {
...
+				expect( stubs.fs.writeJson.getCall( 2 ).args[ 0 ] ).to.equal( '/work/project/packages/ckeditor5-bar/package.json' );
 			} );
 
-			it( 'should update eligible dependencies from the `dependencies` key', () => {
-				stubs.glob.globSync.returns( [ '/work/project/package.json' ] );
+			it( 'should update eligible dependencies from the `dependencies` key', async () => {
+				stubs.glob.glob.resolves( [ '/work/project/package.json' ] );
 
-				stubs.fs.readJsonSync.returns( {
+				stubs.fs.readJson.returns( {
```
Comment (martnpaneq) [inline packages/ckeditor5-dev-release-tools/tests/tasks/updatedependencies.js:185]: *minor*

This probably should be `resolve`, as a promise is expected now.

File: packages/ckeditor5-dev-release-tools/lib/utils/executeinparallel.js (hunk 2 above)
Comment (pomek) [reply packages/ckeditor5-dev-release-tools/lib/utils/executeinparallel.js:56]: I agree, but it's hard to predict what arguments it should receive. I would keep it in mind in the future to refactor it.

**Pull request merged**